### PR TITLE
:book: Update Roadmap with focus on v1alpha3+ (v0.3.x)

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

### CODE REVIEW

File: docs/book/src/roadmap.md [hunk 1]
```diff
@@ -14,18 +14,27 @@ This roadmap is a constant work in progress, subject to frequent revision. Dates
 - [Machine failure remediation](https://github.com/kubernetes-sigs/cluster-api/blob/bf51a2502f9007b531f6a9a2c1a4eae1586fb8ca/docs/proposals/20191030-machine-health-checking.md)
 - [Provider end to end testing framework](https://github.com/kubernetes-sigs/cluster-api/blob/bf51a2502f9007b531f6a9a2c1a4eae1586fb8ca/docs/proposals/20191016-e2e-test-framework.md)
 
-## v0.4 (v1alpha4) ~ July 2020
-
-- [Machine bootstrap failure detection](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/972)
+## v0.3.x (v1alpha3+) ~ July/August 2020
```
Comment (ncdc) [inline docs/book/src/roadmap.md:17]: June please, if possible

Comment (vincepri) [reply docs/book/src/roadmap.md]: Updated

File: docs/book/src/roadmap.md [hunk 2]
```diff
@@ -14,18 +14,27 @@ This roadmap is a constant work in progress, subject to frequent revision. Dates
...
 - [Extensible machine pre-delete hooks](https://github.com/kubernetes-sigs/cluster-api/issues/1514)
-- Define clusterctl inventory specification & have providers implement it
-- Figure out how to change configurations through rolling updates (e.g. apiserver flags, DNS/addons, etc.)
-- [Spot instances support](https://github.com/kubernetes-sigs/cluster-api/issues/1876)
 
-## v0.5 (v1alpha5? v1beta1?) ~ November 2020
+## v0.4 (v1alpha4) ~ Q4 2020
 
+- Define clusterctl inventory specification & have providers implement it
+- Figure out how to change configurations through rolling updates (e.g. apiserver flags, DNS/addons, etc.)
```
Comment (detiber) [inline docs/book/src/roadmap.md:37]: Should we add https://github.com/kubernetes-sigs/cluster-api/issues/2318 to v0.4?

**Comment by detiber** [comment]
> this looks good to me, i do wonder if we should mention the integration with the autoscaler project?

+1, we should likely call out the need for e2e testing of the cluster-api/autoscaler integration

**Comment by elmiko** [comment]
> +1, we should likely call out the need for e2e testing of the cluster-api/autoscaler integration

lol, one step ahead of me Jason ;)

i was going to suggest adding this issue

https://github.com/kubernetes/autoscaler/issues/2942

File: docs/book/src/roadmap.md [hunk 3]
```diff
@@ -14,18 +14,28 @@ This roadmap is a constant work in progress, subject to frequent revision. Dates
...
+- [KCP robustness](https://github.com/kubernetes-sigs/cluster-api/issues/2753)
+- [KCP adoption](https://github.com/kubernetes-sigs/cluster-api/issues/2214)
+- [Clusterctl templating](https://github.com/kubernetes-sigs/cluster-api/issues/2339)
+- [Clusterctl cert-manager lifecycle](https://github.com/kubernetes-sigs/cluster-api/issues/2635)
+- [Post apply YAML experiment](https://github.com/kubernetes-sigs/cluster-api/issues/2395)
+- [MHC Support for control-plane nodes](https://github.com/kubernetes-sigs/cluster-api/issues/2836)
+- [Remote workload cluster watcher](https://github.com/kubernetes-sigs/cluster-api/issues/2414)
 - [Improved status conditions](https://github.com/kubernetes-sigs/cluster-api/issues/1658)
+- [Spot instances support](https://github.com/kubernetes-sigs/cluster-api/issues/1876)
+- [Machine bootstrap failure detection](https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/972)
```
Comment (CecileRobertMichon) [inline docs/book/src/roadmap.md:30]: should this point to https://github.com/kubernetes-sigs/cluster-api/issues/2554 instead?

**Pull request merged**